pkg/apis/autoscaling/v2beta1/conversion.go

```diff
@@ -289,3 +294,46 @@ func Convert_v2beta1_PodsMetricStatus_To_autoscaling_PodsMetricStatus(in *autosc
 	}
 	return nil
 }
+
+func Convert_autoscaling_HorizontalPodAutoscaler_To_v2beta1_HorizontalPodAutoscaler(in *autoscaling.HorizontalPodAutoscaler, out *autoscalingv2beta1.HorizontalPodAutoscaler, s conversion.Scope) error {
+	if err := autoConvert_autoscaling_HorizontalPodAutoscaler_To_v2beta1_HorizontalPodAutoscaler(in, out, s); err != nil {
+		return err
+	}
+	if in.Spec.Behavior != nil {
+		old := out.Annotations
```

Comment:
Why this copy?

Reply:
I copied the implementation from here:
https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/autoscaling/v1/conversion.go#L327

Ok, I'll not use it here.

Reply:
Sorry, I realized that I don't understand the question. Could you rephrase, please?

Reply:
I think @thockin is referring to the statement `old := out.Annotations`. It actually does not copy the map but creates another pointer. Overall this `if` block seems superfluous and can be dropped.

Reply:
Fixed: https://github.com/kubernetes/kubernetes/pull/74525/commits/877d2ffeccb6d8fd1e47a12eab02de44b8bac07b